tests/functional/u/used/used_before_assignment_else_continue.py

```diff
@@ -0,0 +1,42 @@
+"""If the else block continues, it is generally safe to rely on assignments in the except,
+inside the same for loop only."""
+
+
+def safe():
+    """Name used safely inside the loop."""
+    while True:
+        try:
+            pass
+        except ValueError:
+            error = True
+        else:
+            continue
+
+        print(error)
+
+
+def halfway_safe():
+    """Name used safely inside the loop, unsafely outside it."""
+    for _temp in range(0, 1):
+        try:
+            pass
+        except ValueError:
+            error = True
+        else:
+            continue
+
+        print(error)
+    print(error)  # could emit here...
```

Comment:
Is it a TODO or a question for reviewer ? (Is it hard to do ?)

Reply:
It's hard. I can't see a way without a massive refactor. Given that the primer didn't show any effect, I don't think it's terribly important.

however...

I was inspired to work on this because of #1727/#8952, where I saw potentially similar issues with for/if/else/continue. Turns out this current fix only works for for/try/except/else/continue. :-(

Reply:
```suggestion
    print(error)  # Should raise used-before-assignment see #XXXX...
```
We need to create the issue though. Lazy version:
```suggestion
    print(error)  # Should raise used-before-assignment see https://github.com/pylint-dev/pylint/pull/9374#discussion_r1460984807
```

Reply:
I'll do the non-lazy version.